### PR TITLE
Fixing issue for reference allele

### PR DESCRIPTION
**Pull request by luissian**
Fixing issue with long files pandas at reference allele function
Starting defining alleles for allele calling

### CODE REVIEW

File: taranis/distance.py [hunk 1]
```diff
@@ -72,7 +72,13 @@ def create_matrix(self) -> pd.DataFrame:
         dist_matrix.write("alleles\t" + "\t".join(allele_names) + "\n")
         dist_matrix.write("\n".join(out_data[1:]))
         dist_matrix.seek(0)
-        matrix_pd = pd.read_csv(dist_matrix, sep="\t", index_col="alleles").fillna(0)
+        file_test = "/media/lchapado/Reference_data/proyectos_isciii/taranis/test/reference_alleles_testing_full_schema_17a/error.csv"
```
Comment (saramonzon) [inline taranis/distance.py]: should this be here?

File: taranis/__main__.py [hunk 2]
```diff
@@ -442,7 +449,8 @@ def allele_calling(
     """
     results = []
     for assembly_file in assemblies:
-        results.append(taranis.allele_calling.parallel_execution(assembly_file, schema, schema_ref_files, output))
-
+        assembly_name = Path(assembly_file).stem
+        results.append({assembly_name: taranis.allele_calling.parallel_execution(assembly_file, schema, schema_ref_files, output)})
+        pdb.set_trace()
```
Comment (saramonzon) [inline taranis/__main__.py]: missing pdb.set_trace()

File: .github/workflows/pytest.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    runs-on: ubuntu-latest
+  
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.10
```
Comment (saramonzon) [inline .github/workflows/pytest.yml]: something is failing here in the test

File: taranis/__main__.py [hunk 4]
```diff
@@ -15,6 +15,9 @@
 import taranis.reference_alleles
 import taranis.allele_calling
 
+from pathlib import Path
+import pdb
```
Comment (saramonzon) [inline taranis/__main__.py]: should we import pdb and commit it?

File: taranis/allele_calling.py [hunk 5]
```diff
@@ -23,105 +25,142 @@
...
+                    result["allele_match"][allele_name],
+                    result["allele_details"][allele_name],
+                ) = self.assign_allele_type(blast_result, allele_file, allele_name)
+                # pdb.set_trace()
+            else:
+                # Sample does not have a reference allele to be matched
+                # Keep LNF info
+                # ver el codigo de espe
+                # lnf_tpr_tag()
+                pdb.set_trace()
```
Comment (saramonzon) [inline taranis/allele_calling.py]: another pdb, i guess this is testing you are doing now

File: taranis/utils.py [hunk 6]
```diff
@@ -255,6 +257,31 @@ def get_files_in_folder(folder: str, extension: str = None) -> list[str]:
...
+    """
+    try:
+        result = subprocess.run(
+            ["grep", parameters, pattern, input_file],
+            capture_output=True,
+            check=True,
+            text=True,
+        )
+    except subprocess.CalledProcessError as e:
+        pdb.set_trace()
```
Comment (saramonzon) [inline taranis/utils.py]: another pdb

File: .github/workflows/pytest.yml (hunk 3 above)
Comment (luissian) [reply .github/workflows/pytest.yml]: Fixed

File: taranis/__main__.py (hunk 4 above)
Comment (luissian) [reply taranis/__main__.py]: all pdb's are removed from code

File: taranis/allele_calling.py (hunk 5 above)
Comment (luissian) [reply taranis/allele_calling.py]: removed

File: taranis/utils.py (hunk 6 above)
Comment (luissian) [reply taranis/utils.py]: Same removed

**Review by saramonzon — APPROVED** [review]

**Pull request merged**